iceoryx_posh/test/integrationtests/test_popo_chunk_building_blocks.cpp

```diff
@@ -54,7 +54,7 @@ struct ChunkDistributorConfig
 
 struct ChunkQueueConfig
 {
-    static constexpr uint64_t MAX_QUEUE_CAPACITY = NUM_CHUNKS_IN_POOL;
+    static constexpr uint64_t MAX_QUEUE_CAPACITY = iox::MAX_SUBSCRIBER_QUEUE_CAPACITY;
```

Comment:
@budrus @mossmaurice can you please confirm that this makes sense?

Reply:
I feel that this was chosen for a reason. With `NUM_CHUNKS_IN_POOL`, we avoid queue overflows (which are also checked in the test and lead to fail). I think we have timing issues with the Mac sanitize builds. If the forward and subscriber thread are not fast enough you either have a queue overflow or a mempool running out of chunks error. With this change you get the latter one but it still fails if the threads get no runtime

Reply:
but then we exhaust the mempool, since there is
- `lastChunk` in publisher
- two queues which can hold `NUM_CUNKS_IN_POOL`

The overflow problem was solved by using a FIFO instead of a SOFI

Reply:
When the other threads are not fast enough it will break the one or the other way. You can now configure how it fails. SOFI or FIFO is also don't care as we should set the overflow flag also for SOFI and the test fails if there is on overflow

Reply:
you're right. But this makes my PR green :D

Reply:
yesterday two other PRs had this problem. It seems they are building now. I will revert it and check if it builds now. If not, we should try for some time with a smaller queue size

Reply:
okay, I will revert my revert. The original test now fails again

Reply:
You are hunting a ghost. Both versions will fail. It's just gambling. We have to change the test or somehow disable it in the sanitize build. I'm pretty sure that it is a timing issue

Reply:
you are totally right, but for some reasons I had 4 red builds without this change and 2 greed builds with this change ... is Vegas still open?